tests/system.rs

```diff
@@ -485,3 +503,142 @@ fn string_eq_ignore_newline_repr_test() {
     assert!(string_eq_ignore_newline_repr("a\r\n\r\n\r\nb", "a\n\n\nb"));
     assert!(!string_eq_ignore_newline_repr("a\r\nbcd", "a\nbcdefghijk"));
 }
+
+struct ConfigPair {
+    name: String,
+    value: String,
+}
+
+struct CodeBlock {
+    config: ConfigPair,
+    code_block: String,
+    code_block_start_line: u32,
+}
+
+// Read Configurations.md and build a `Vec` of `CodeBlock` structs with one
+// entry for each Rust code block found. Behavior:
+// - Rust code blocks are identifed by lines beginning with "```rust".
+// - One explicit configuration setting is supported per code block.
+// - Rust code blocks with no configuration setting are illegal and cause an
+//   assertion failure.
+// - Configuration names in Configurations.md must be in the form of
+//   "## `NAME`".
+// - Configuration values in Configurations.md must be in the form of
+//   "#### `VALUE`".
+fn get_code_blocks() -> Vec<CodeBlock> {
+    let filename = "Configurations.md";
+    let filebuf = BufReader::new(
+        fs::File::open(filename).expect(&format!("Couldn't read file {}", filename)),
+    );
+
+    let config_name_regex =
+        regex::Regex::new(r"^## `([^`]+)`").expect("Failed creating configuration pattern");
+    let config_value_regex = regex::Regex::new(r#"^#### `"?([^`"]+)"?`"#)
+        .expect("Failed creating configuration value pattern");
+
+    let mut code_blocks = Vec::new();
```

Comment:
I think this whole parsing function could be nicely refactored into a struct and a set of functions, rather than being a big function with lots of local, mutable state.